workflows/walletsTracker.go

```diff
@@ -0,0 +1,163 @@
+// processLoop go routine that can be stopped by calling the Close method
+func NewWalletTracker(args WalletTrackerArgs) (*walletTracker, error) {
+	if check.IfNil(args.TrackableAddressesProvider) {
+		return nil, ErrNilTrackableAddressesProvider
+	}
+	if check.IfNil(args.Proxy) {
+		return nil, ErrNilProxy
+	}
+	if check.IfNil(args.NonceHandler) {
+		return nil, ErrNilLastProcessedNonceHandler
+	}
+	if args.MinimumBalance == nil {
+		return nil, ErrNilMinimumBalance
+	}
+
+	wt := &walletTracker{
+		accumulator:                newAddressesAccumulator(),
+		trackableAddressesProvider: args.TrackableAddressesProvider,
+		proxy:                      args.Proxy,
+		nonceHandler:               args.NonceHandler,
+		checkInterval:              args.CheckInterval,
+		minimumBalance:             args.MinimumBalance,
+	}
+
+	var ctx context.Context
+	ctx, wt.cancelFunc = context.WithCancel(context.Background())
+	go wt.processLoop(ctx)
+
+	return wt, nil
+}
+
+func (wt *walletTracker) processLoop(ctx context.Context) {
+	log.Debug("walletTracker.processLoop started")
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Debug("terminating walletTracker.processLoop...")
+			return
+		case <-time.After(wt.checkInterval):
+			err := wt.fetchAndProcessHyperBlocks()
+			log.LogIfError(err)
+		}
+	}
+}
+
+func (wt *walletTracker) fetchAndProcessHyperBlocks() error {
+	lastProcessedNonce := wt.nonceHandler.GetLastProcessedNonce()
+	networkNonce, err := wt.proxy.GetLatestHyperBlockNonce()
+	if err != nil {
+		return err
+	}
+
+	for nonce := lastProcessedNonce + 1; nonce <= networkNonce; nonce++ {
+		err = wt.fetchAndProcessHyperBlock(nonce)
+		if err != nil {
+			return err
+		}
+
+		wt.nonceHandler.ProcessedNonce(nonce)
+	}
+
+	return nil
+}
+
+func (wt *walletTracker) fetchAndProcessHyperBlock(nonce uint64) error {
+	block, err := wt.proxy.GetHyperBlockByNonce(nonce)
+	if err != nil {
+		return err
+	}
+
+	wt.processHyperBlock(block)
+
+	log.Debug("processed hyper block", "nonce", nonce, "hash", block.Hash, "num txs", block.NumTxs)
+
+	return nil
+}
+
+func (wt *walletTracker) processHyperBlock(block *data.HyperBlock) {
+	for _, transaction := range block.Transactions {
+		err := wt.processTransaction(transaction)
+		if err != nil {
+			transactionString, _ := json.Marshal(&transaction)
+			log.Warn("error processing transaction, ignoring",
+				"transaction", transactionString, "error", err)
+		}
+	}
+}
+
+func (wt *walletTracker) processTransaction(transaction data.TransactionOnNetwork) error {
+	value, ok := big.NewInt(0).SetString(transaction.Value, 10)
+	if !ok {
+		return ErrInvalidTransactionValue
+	}
+
+	if !wt.trackableAddressesProvider.IsTrackableAddresses(transaction.Receiver) {
+		return nil
+	}
+
+	if value.Cmp(wt.minimumBalance) < 0 {
+		// transaction has a very small value transfer (possible attack vector as someone
+		//can trigger millions of these transactions as to consume the owner's balance through fees)
+		return nil
+	}
+
```

Comment:
might inject here also a component (or even easier - call a function) that does something with the transaction. in your case, it can be a simple parser, but it would be really nice to show this function as a placeholder for interactions with the relevant transactions

Reply:
added a handler function (thought about a notifier interface but looked overkill)